schema/schema_test.go

```diff
@@ -0,0 +1,364 @@
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "override template",
+			Input: map[string]string{
+				"test.tf": `
+resource "aws_instance" "web" {
+  ami           = "ami-b73b63a0"
+  instance_type = "t1.2xlarge" # invalid type!
+
+  tags {
+    Name = "HelloWorld"
+  }
+}
+`,
+				"test_override.tf": `
+resource "aws_instance" "web" {
+  ami           = "ami-override"
+  instance_type = "t2.nano"
+
+  tags {
+    Version = "0.1"
+  }
+}
+`,
+				"override.tf": `
+resource "aws_instance" "web" {
+  instance_type = "t2.micro"
+}
+`,
+			},
+			Result: []*Template{
+				{
+					File: "test.tf",
+					Resources: []*Resource{
+						{
+							File: "test.tf",
+							Type: "aws_instance",
+							Id:   "web",
+							Pos: token.Pos{
+								Filename: "test.tf",
+								Offset:   31,
+								Line:     2,
+								Column:   31,
+							},
+							Attrs: map[string]*Attribute{
+								"ami": {
+									Poses: []token.Pos{
+										{
+											Filename: "test_override.tf",
+											Offset:   51,
+											Line:     3,
+											Column:   19,
+										},
+									},
+									Vals: []interface{}{
+										token.Token{
+											Type: 9,
+											Pos: token.Pos{
+												Filename: "test_override.tf",
+												Offset:   51,
+												Line:     3,
+												Column:   19,
+											},
+											Text: "\"ami-override\"",
+											JSON: false,
+										},
+									},
+								},
+								"instance_type": {
+									Poses: []token.Pos{
+										{
+											Filename: "test_override.tf",
+											Offset:   84,
+											Line:     4,
+											Column:   19,
+										},
+									},
+									Vals: []interface{}{
+										token.Token{
+											Type: 9,
+											Pos: token.Pos{
+												Filename: "test_override.tf",
+												Offset:   84,
+												Line:     4,
+												Column:   19,
+											},
+											Text: "\"t2.nano\"",
+											JSON: false,
+										},
+									},
+								},
+								"tags": {
+									Poses: []token.Pos{
+										{
+											Filename: "test_override.tf",
+											Offset:   102,
+											Line:     6,
+											Column:   8,
+										},
+									},
+									Vals: []interface{}{
+										map[string]interface{}{
+											"Version": token.Token{
+												Type: 9,
+												Pos: token.Pos{
+													Filename: "test_override.tf",
+													Offset:   118,
+													Line:     7,
+													Column:   15,
+												},
+												Text: "\"0.1\"",
+												JSON: false,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		files := map[string][]byte{}
+		for filename, body := range tc.Input {
+			files[filename] = []byte(body)
+		}
+		templates, err := Make(files)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, template := range templates {
+			sort.Slice(template.Resources, func(i, j int) bool {
```

Comment:
__[gometalinter]__
* [gosimple] Slice not declared by package sort
* [staticcheck] Slice not declared by package sort
* [varcheck] unused variable or constant Slice not declared by package sort

<!-- Comment by [SideCI](https://sideci.com) -->